### PR TITLE
fix: issues with running HTTPS requests through HTTP proxy

### PR DESCRIPTION
**Pull request by edvald**
**What this PR does / why we need it**:

This adds and configured `global-agent`
(https://github.com/gajus/global-agent) and replaces Axios with `got` 
(https://github.com/sindresorhus/got) which, since Axios has a
long-standing issue with routing HTTPS requests through HTTPS proxies.

**Which issue(s) this PR fixes**:

Reported by @khaled on Slack

### CODE REVIEW

File: garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts [hunk 1]
```diff
@@ -126,12 +126,12 @@ async function getImagesInRegistry(ctx: KubernetesPluginContext, log: LogEntry)
 
   while (nextUrl) {
```
Comment (thsig) [inline garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts]: Maybe we should extract this `while` loop(which is mostly repeated below) into a helper function?

**Comment by thsig** [comment]
Should we globally replace Axios with `got`? We're still using it in a couple of places (one in the dashboard, and in `support/support-util.ts`.

File: garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts (hunk 1 above)
Comment (edvald) [reply garden-service/src/plugins/kubernetes/commands/cleanup-cluster-registry.ts]: This is not related to the PR, per se, but yeah I guess so.

**Comment by edvald** [comment]
We could, sure. It doesn't really matter for those cases, except for consistency, but that has its own merit.

**Pull request merged**